### PR TITLE
CLOUDP-114068: [mcli] drop EOL OSes

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-114068

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by andreaangiolillo — DISMISSED** [review]
LGTM

File: build/ci/release.yml [hunk 1]
```diff
@@ -339,17 +339,6 @@ tasks:
         vars:
           image: centos7-rpm
           tool_name: mongocli
-  - name: pkg_test_mongocli_docker_centos8
```
Comment (gssbzn) [inline build/ci/release.yml:342]: centos8 is EOL since January the image is unstable now

File: build/ci/release.yml [hunk 2]
```diff
@@ -599,28 +588,6 @@ tasks:
           distro: rhel70
           ext: rpm
           tool_name: atlascli
-  - name: push_atlascli_rhel62_org
```
Comment (gssbzn) [inline build/ci/release.yml:602]: rhel 6 has been EOL for a while, time to remove it

**Review by tibulca — APPROVED** [review]

**Review by andreaangiolillo — APPROVED** [review]
Could you share the file that contains the supported distro? It would be great to add its link  somewhere in mongocli so that we can add/remove distros

**Comment by gssbzn** [comment]
> Could you share the file that contains the supported distro? It would be great to add its link somewhere in mongocli so that we can add/remove distros

good ask, shared internally

**Pull request merged**